learn/guides/calling-java-code-from-ballerina.md

```diff
@@ -0,0 +1,290 @@
+
+You can see them in the imported class list. We encourage you to generate Ballerina bindings for these four classes as a start.  
+
+Now, we'll create an environment for our Ballerina program. 
+
+### Step 2 - Setting Up the Ballerina Package
+
+This section assumes that you have already read the [Package Layout](/learn/user-guide/ballerina-packages/) guide.
+
+#### Creating a Ballerina Package
+```sh
+> bal new yaml_package
+Created new Ballerina package 'yaml_package' at yaml_package.
+```
+
+#### Verifying the Package
+```sh
+> cd yaml_package
+> bal build
+Compiling source
+	sameera/yaml_package:0.1.0
+
+Generating executable
+	target/bin/yaml_package.jar
+```
+```sh
+> bal run target/bin/yaml_package.jar
+Hello World!
+```
+Great! You are all set for the next step.
+
+#### Adding a Sample YAML File 
+Copy the below content to a file named `invoice.yml` in the package root directory.
+```yaml
+invoice: 34843
+date   : 2001-01-23
+bill-to: &id001
+   given  : Chris
+   family : Dumars
+   address:
+       lines: |
+           458 Walkman Dr.
+           Suite #292
+       city    : Royal Oak
+       state   : MI
+       postal  : 48046
+ship-to: *id001
+product:
+   - sku         : BL394D
+     quantity    : 4
+     description : Basketball
+     price       : 450.00
+   - sku         : BL4438H
+     quantity    :
+     description : Super Hoop
+     price       : 2392.00
+tax  : 251.42
+total: 4443.52
+comments: >
+   Late afternoon is best.
+   Backup contact is Nancy
+   Billsmer @ 338-4338.\
+```
+
+### Step 3 - Generating the Ballerina Bindings 
+In this step, we'll use the `bindgen` tool to generate Ballerina bindings for those four classes that we talked about in Step 1. If you want more information about the tool, you can refer [The `bindgen` tool](#the-bindgen-tool) section.
+
+```sh
+> bal bindgen -mvn org.yaml:snakeyaml:1.25 org.yaml.snakeyaml.Yaml java.io.FileInputStream java.io.InputStream java.util.Map
+
+Ballerina package detected at: /Users/sameera/yaml_package
+
+Resolving maven dependencies...
+snakeyaml-1.25.jar 100% [===============================================] 297/297 KB (0:00:01 / 0:00:00)
+
+Updated the `Ballerina.toml` file with the new platform libraries.
+
+The following JARs were added to the classpath:
+	snakeyaml-1.25.jar
+
+Generating bindings for:
+	java.util.Map
+	java.io.FileInputStream
+	org.yaml.snakeyaml.Yaml
+	java.io.InputStream
+
+Generating dependency bindings for:
+	org.yaml.snakeyaml.introspector.BeanAccess
+	java.util.function.BiFunction
+	org.yaml.snakeyaml.DumperOptions$FlowStyle
+	...
+	...
+```
+
+- The `-mvn` option specifies the Maven dependency of the Java library required to generate bindings.
+- The argument list specifies the Java class names. 
+
+The `bindgen` tool generates bindings for:
+- The specified Java classes.
+- The Java classes exposed in the public APIs of all the specified classes.
+
+
+Before we move onto the next step, let’s verify the generated code. 
```

Comment:
```suggestion
Before we move onto the next step, let’s verify the generated code. 
```
Shall we make this also a step?

Reply:
It's not exactly another step, it's just doing a normal build to verify what you have done.